### ISSUE TITLE
IDE0031: false positive when identifier is ambiguous between variable and type

### ISSUE THREAD

**Issue by sodiboo** [dotnet/roslyn#82281]
**Version Used**: 

.NET SDK `10.0.102` with `Compiler version: '5.0.0-2.26055.108 (1def04f3)'. Language version: 14.0.`

**Steps to Reproduce**:

```cs
// Program.cs
var foo = new Foo()
{
    Format = "Hello from the {0} method",
};

Console.WriteLine(IncorrectFix(foo));

static string? IncorrectFix(Foo? Foo) => Foo is null ? null : Foo.HelloFormat(Foo.Format);

class Foo
{
    public required string Format { get; init; }
    public string HelloFormat(params string[] _) => string.Format(Format, "instance");
    public static string HelloFormat(string Format) => string.Format(Format, "static");
}
```
```xml
<!-- Example.csproj -->
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net10.0</TargetFramework>
    <OutputType>exe</OutputType>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>
</Project>
```
```sh
$ dotnet format style --severity info --verify-no-changes
./Program.cs(9,42): info IDE0031: Null check can be simplified [./Example.csproj]
$ dotnet run
Hello from the static method
$ dotnet format style --severity info
$ dotnet run
Hello from the instance method
```
**Diagnostic Id**: `IDE0031: Null check can be simplified`

**Expected Behavior**:
The style fix should not be suggested.

**Actual Behavior**:
The style fix is suggested, and can be automatically applied, changing the meaning of the code.

When i ran into this in an actual codebase, it actually just led to a compile error. But in the example above, the code still compiles, and gives a *different result*.

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/63557

**Comment by sodiboo** [comment]
Good bot. That is actually the same issue. But it is still present in modern Roslyn, apparently. 😔
